### DISCUSSION TITLE
Document that the order of Rule.user_classes has no meaning

### DISCUSSION THREAD

**Discussion by zanebclark** [openmobilityfoundation/curb-data-specification#82] [Data Specification]
[Rule](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/curbs#rule).`user_classes` is an Array, so ["a", "b"] != ["b", "a"]. We should either a) give the order of `user_classes` meaning or b) make explicit that the Array should be interpreted as a set and ["a", "b"] should be interpreted to be equivalent to ["b", "a"].

**Comment by schnuerle** [comment]
Ah I see now about making the order here explicit. I think we can give the order of `user_classes` meaning, that the first one takes priority over later ones.

**Reply by zanebclark** [reply]
I'm not sure it would address the issue we discussed in the meeting earlier today, as the two "overlapping" policies wouldn't share a `user_classes` array

**Reply by jacobmalleau** [reply]
I think we should make the order not matter. It should purely be if it is in the array or not. I am not sure how you even specificy priority of a "permit" over a "truck". If anything, this priority would depend on the policy itself.

**Comment by zanebclark** [comment]
A further clarification on `user_classes`: 
>If specified, this regulation only applies to users matching the user classes contained within. If not specified, this regulation applies to everyone.

By ` matching the user classes contained within`, do all user classes need to match? Or, if only one user class matches, the policy applies?

**Reply by jacobmalleau** [reply]
This may be messy, but I think the rule is different depending which way you look at it.

Users Looking Up Policies: as long as at least one user class of the user is in the policy, that policy applies to the user.
For example, you can have policies that apply just to "trucks" as well as "trucks" with "loading permit"s.
So a user that is a truck with a loading permit, would want to know about both policies for ["trucks", "loading permits"] and ["trucks"]

Policies Defined By User Classes: all user classes must be defined by the user for the policy to apply.
For example: all policies defined by ["loading permits, "trucks"] are different than policies defined by ["trucks"] or ["loading permits"].

Ultimately, I dont know how much this matters. I think a city can set up the query whichever way they want. 
For your statement above, I think it should be "at least one user class must match for the policy to apply".

Hopefully this doesnt confuse things more.

**Comment by schnuerle** [comment]
I added this text, which is my best attempt at synthesizing the conversations here. 

> The order of `user_classes` is not relevant, but a vehicle using a curb with this rule must match all `user_classes` contained in the array.

Welcome your thoughts on adjusting and if this works for you all!

https://github.com/openmobilityfoundation/curb-data-specification/blob/release-1.0.0/curbs/README.md#rule
